sensor-validate.c

```diff
@@ -1,28 +1,177 @@
+/*****************************************************************************/
+/***    \file        sensor-validate.c
+ ***    \author      Gagandeep Gawadia
+ ***
+ ***    \brief       File contains function to evaluate sensor state 
+******************************************************************************/
+
+/*-------------------------------  Include files  -------------------------------------------*/
 #include "sensor-validate.h"
+#include <cstddef>
+#include <cstdio>
+
+/*-------------------------------  Macro definition ----------------------------------------- */
+#define soc_delta 0.05f
+#define currentreading_delta 0.1f
+
+/* -------------------------------- Function definition--------------------------------------- */
+
 
-int _give_me_a_good_name(double value, double nextValue, double maxDelta) {
+/*****************************************************************************
+Function name:  NospikeIsDetected                                        *//*!
+\brief          Evaluates if spike is detected with consecutive dataset
+\return         true, if spike is not detected
+                false otherwise
+*//**************************************************************************/
+
+
+int NospikeIsDetected(double value, double nextValue, double maxDelta) {
   if(nextValue - value > maxDelta) {
     return 0;
   }
   return 1;
 }
 
-int validateSOCreadings(double* values, int numOfValues) {
-  int lastButOneIndex = numOfValues - 1;
-  for(int i = 0; i < lastButOneIndex; i++) {
-    if(!_give_me_a_good_name(values[i], values[i + 1], 0.05)) {
-      return 0;
-    }
-  }
-  return 1;
+/*****************************************************************************
+Function name:  evaluationofspikedetected                            *//*!
+\brief          Evaluates if spike is detected with provide input array set
+\return         false, if spike is detected
+                true otherwise
+*//**************************************************************************/
+
+int evaluationofspikedetected(double* values, int numOfValues, float deltadifference )
+{
+	int lastButOneIndex = numOfValues - 1;
+	for (int i = 0; i < lastButOneIndex; i++) {
+		if (!NospikeIsDetected(values[i], values[i + 1], deltadifference)) {
```

Comment:
When you encounter a double negative: the negation operator `!` and a function starting with `No` - see if you can remove the double negative.